Katello/katello
# Fixes #36453 - Prevent regenerating metadata for repositories that use complete mirroring

#### What are the changes introduced in this pull request?
1. Deprecates --force flag on repository republish metadata and CV version metadata regenerate.
2. Errors out for republishing metadata on complete mirroring repositories.
3. Skips Complete Mirroring repositories of a CV version when republishing metadata of version repos
#### Considerations taken when implementing this change?
On the UI, we disable regenerate metadata button for 'Complete Mirroring' repos and CV versions with any 'Complete Mirroring' repositories.
On the API and hammer, we now fail on republishing metadata on 'Complete Mirroring' repos and 'Complete Mirroring' repositories in a CV version will be skipped.
#### What are the testing steps for this pull request?
1. Create a couple repos, one with Complete Mirroring and another with content only or additive.
2. In hammer, try hammer -r repository republish --id=`id` for both. 
3. Add both repos to a CV and publish a version.
4. In hammer, try hammer -r content-view version republish-repositories --id=`id` . Notice that the foreman task only republishes metadata on the version repos for the non-complete mirroring repo.
